src/app/components/Navbar.tsx

```diff
@@ -0,0 +1,253 @@
+"use client";
+
+import React, { useState, useEffect, useRef } from "react";
+import Link from "next/link";
+import { usePathname } from "next/navigation";
+
+const languages = [
+  { label: "English", value: "en" },
+  { label: "සිංහල", value: "si" },
+  { label: "தமிழ்", value: "ta" },
+];
+
+const Navbar = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [selectedLanguage, setSelectedLanguage] = useState("en");
+  const pathname = usePathname();
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const toggleMenu = () => setIsOpen(!isOpen);
+  const closeMenu = () => setIsOpen(false);
+
+  const toggleDropdown = () => setIsDropdownOpen(!isDropdownOpen);
+  const closeDropdown = () => setIsDropdownOpen(false);
+
+  const handleLanguageSelect = (lang: string) => {
+    setSelectedLanguage(lang);
+    closeDropdown();
+  };
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        closeDropdown();
+      }
+    };
+
+    const handleEscape = (event: KeyboardEvent) => {
```

Comment:
When tested, the escape key did not close the menu. But maybe we'll remove this since it won't be much useful 🤔